### DISCUSSION TITLE
Picardで出てくるエラーについて

### DISCUSSION THREAD

**Discussion by sairyosw** [nig-sc/nigsc_homepage2#313] [Q&A] [answered]
export MALLOC_ARENA_MAX=2
export _JAVA_OPTIONS="-Xmx50g"

for name in `cat name.list`
do
     indexp=/usr/local/biotools/p/picard:2.27.4--hdfd78af_0
     singularity exec $indexp picard AddOrReplaceReadGroups \
     -I /home/swmugi/bam/${name}_sorted_uniq.bam \
     -O /home/swmugi/bam/${name}_sorted_uniq_pic.bam \
     -RGID $name -RGLB library -RGPL IONTORRENT -RGPU IonS5 -CREATE_INDEX true -RGSM $name 

done

上記のようなコードをqsubで計算した結果、
[Tue Mar 28 03:09:46 GMT 2023] picard.sam.AddOrReplaceReadGroups done. Elapsed time: 0.01 minutes.
Runtime.totalMemory()=2147483648
/usr/local/bin/picard: line 5: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
Picked up _JAVA_OPTIONS: -Xmx50g
03:09:47.546 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/local/share/picard-2.27.4-0/picard.jar!/com/intel/gkl/native/libgkl_compression.so
[Tue Mar 28 03:09:47 GMT 2023] AddOrReplaceReadGroups --INPUT /home/swmugi/bam/BACB001A03_sorted_uniq.bam --OUTPUT /home/swmugi/bam/BACB001A03_sorted_uniq_pic.bam --RGIDサンプル名 --RGLB library --RGPL IONTORRENT --RGPU IonS5 --RGSM サンプル名--CREATE_INDEX true --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
[Tue Mar 28 03:09:47 GMT 2023] Executing as swmugi@igt004 on Linux 3.10.0-1160.el7.x86_64 amd64; OpenJDK 64-Bit Server VM 11.0.15-internal+0-adhoc..src; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT
INFO	2023-03-28 03:09:48	AddOrReplaceReadGroups	Created read-group ID=サンプル名 PL=IONTORRENT LB=library SM=サンプル名
のような結果が帰ってきました。
/usr/local/bin/picard: line 5: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
の部分が気になるのですが、このエラーを解消するためにはスパコンの設定を変更しなければいけないようです。
解決策をご存じの方がいらっしゃいましたら、教えてください。

**Comment by nigyta** [comment]
表示されている警告は時刻や地域に関するもので解析結果には影響しないため、無視しても良いと思います。
LANGやLC_ALLなどの環境変数を適切に設定すれば表示されないようにできそうですが、いくつか設定を試してみた限りではうまくいきませんでした。

**Reply by sairyosw** [reply, marked as answer]
お忙しい中、ご返答いただきありがとうございます。
特に、気にする必要がないとのことで、とても安心しました。
今後ともよろしくお願いします。
